src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -811,8 +811,23 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
 
     private async handleDeletePictureClick() {
         try {
-            await apiService.flagDeleteImage(3 /* this.state.selectedAsset.asset.id */);
-            await apiService.deleteImage(3 /* this.state.selectedAsset.asset.id */);
+            const { auth, trackingActions } = this.props;
+            const { selectedAsset, assets } = this.state;
+            const indexAssetToRemove = assets.findIndex((asset: IAsset) => {
+                return asset.id === selectedAsset.asset.id;
+            });
+
+            await trackingActions.trackingImgDelete(auth.userId, selectedAsset.asset.id);
+            assets.splice(indexAssetToRemove, 1);
```

Comment:
Try to avoiding mutating objects. You might have later problems with debugging ;) 

https://jaketrent.com/post/remove-array-element-without-mutating/